### PR TITLE
[WIP/RFC] Fix yaml indent: look at previous mapping keys

### LINKED ISSUE #11039: yaml file indent issue
This should be a general issue and also appears in vim, my neovim version is:

```
NVIM v0.3.8
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20190704-11886-1ukicoc/neovim-0.3.8/build/config -I/tmp/neovim-20190704-11886-1ukicoc/neovim-0.3.8/src -I/usr/local/include -I/usr/local/opt/gettext/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include -I/tmp/neovim-20190704-11886-1ukicoc/neovim-0.3.8/build/src/nvim/auto -I/tmp/neovim-20190704-11886-1ukicoc/neovim-0.3.8/build/include
Compiled by brew@Mojave.local

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.3.8/share/nvim"

Run :checkhealth for more info
```

### Steps to reproduce using `nvim -u NORC`

1. Create a `test.yaml` file with following content(include the last two blank lines):

```
branches:
  only:
    - master


```

2. Open this file with `nvim test.yaml`, scroll to bottom of this file and add a new key-pair.

vim-yaml-bug

### Actual behaviour

The new key-value pair becomes the child element of the previous one.

### Expected behaviour

The new key-value pair should be top level element.

## Root Cause

The reason is because of this file: https://github.com/neovim/neovim/blob/master/runtime/indent/yaml.vim

When I remove this file, I get the expected result.

Stackoverflow: https://stackoverflow.com/questions/57951841/vim-neovim-yaml-indent-bug

cc @justinmk @dbarnett @KillTheMule

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neovim/neovim/issues/11039.

Not tested much, but added an indent test for it already.

As per https://github.com/neovim/neovim/issues/11039#issuecomment-532131153 the file is not maintained currently externally, due to inactivity of @ZyX-I.

### CODE REVIEW

**Review by chenzhiwei — APPROVED** [review]
This fix works for me.

**Comment by blueyed** [comment]
@chenzhiwei 
Thanks for the feedback.

It really looks like https://github.com/vim/vim/commit/dc1f1645cb495fa6bfbe216d7359f23539a0e25d#diff-79d61bef04e12397c4a5d0494a590d30R40 might have been done by mistake.
It might even make sense to remove this condition (check/ignore comments).

The test failure is due to something that was broken before already IIRC - the test should be marked pending probably.

**Comment by chenzhiwei** [comment]
@chrisbra @k-takata Could you help review and merge this PR?

**Comment by k-takata** [comment]
@brammool It seems that the maintainer of this file (@Zyx-I) is currently inactive. So, could you consider merging this?

**Comment by blueyed** [comment]
I do not consider this to be ready.
The last added test should have failed AFAICS (likely does now after rebase, via cd67059).
And it is not clear why the change this reverts was done in the first place, and why comments are handled in a special way there.

**Comment by chrisbra** [comment]
so do you intend to finish this up?

**Comment by blueyed** [comment]
> so do you intend to finish this up?

Not currently - it is e.g. blocked on feedback from @brammool with regard to https://github.com/vim/vim/pull/4946#issuecomment-532498759.

I think it's good to have tests for this to start with though, of course.

So I suggest backing out the failing test again (which fails without the patch already), and then use the minimal patch (2nd commit).

But since there are no other tests yet I am not sure about regressions etc.

Not planning to push it here, since it does not get merged anyway directly.

**Comment by brammool** [comment]
Not sure what input is needed from me.  Can we just fix the problem?
What happened with Nikolai, he used to be quite active.

**Comment by blueyed** [comment]
> Not sure what input is needed from me.

It looked like it might have been updated/changed accidentally.  But the patch was from Nikolai then maybe - cannot really be seen anymore, also there's no specific commit for it, but it was with a generic update-runtime commit (IIRC).  Having real VCS history (and authorship) for this would be useful. I don't know what happened to him.

>  Can we just fix the problem?

It is fixed.  There are two variants, one being the simple revert - after I've noticed that there is generic code doing what I've done explicitly in the first case already.

I've not used it much since then myself though, and since it reverts something it might cause a regression.

Feel free to take the simple change + the first test, and then we can iterate from there based on user reports, adding tests then.

**Comment by brammool** [comment]
When I include the change to yaml.vim then the test fails: the last line with "map: val" is not indented.

**Comment by blueyed** [comment]
The test added in 5b5ec9d59 fails also on master already.

**Comment by chrisbra** [comment]
included as of 2e693a88b24dc6b12883fad78ff2cb9cd4469c98

**Pull request closed without merging**
